src/interface/swarm.hpp

```diff
@@ -42,36 +45,78 @@ class SwarmDeviceContext {
   KOKKOS_FUNCTION
   bool IsActive(int n) const { return mask_(n); }
 
+  KOKKOS_FUNCTION
+  bool IsOnCurrentMeshBlock(int n) const { return blockIndex_(n) == this_block_; }
+
   KOKKOS_FUNCTION
   void MarkParticleForRemoval(int n) const { marked_for_removal_(n) = true; }
 
```

Comment:
make n const

Reply:
this function is gone and is now just an extra reference argument to `GetNeighborBlockIndex` to avoid the confusion of requiring a separate call to that function to use `IsOnCurrentMeshBlock`